lib/Dialect/Torch/Transforms/RefinePublicReturn.cpp

```diff
@@ -71,7 +71,10 @@ class RefinePublicReturnPass
           // If the return (or transitively other ops) are not the only users,
           // then we can't be sure that the tensor hasn't been mutated, so stop
           // here.
-          if (!llvm::hasSingleElement(copy->getUsers()))
+          SmallVector<Operation *> usersVector(copy->getUsers());
```

Comment:
I think this is buggy if there are noncontiguous uses like one use by return, one use by other, another use by return.

Perhaps `SetVector<Operation *> users(copy->getUsers().begin(), copy->getUsers().end); if (users.size() == 1) ...`

Reply:
Done!